repos/system_upgrade/common/actors/reportleftoverpackages/reportleftoverpackages/tests/test_reportleftoverpackages.py

```diff
@@ -0,0 +1,57 @@
+from leapp import reporting
+from leapp.libraries.actor import reportleftoverpackages
+from leapp.libraries.common.testutils import create_report_mocked, CurrentActorMocked, logger_mocked
+from leapp.libraries.stdlib import api
+from leapp.models import LeftoverPackages, RemovedPackages, RPM
+
+
+def test_no_leftover_and_no_removed_packages(monkeypatch):
+    monkeypatch.setattr(api, 'current_actor', CurrentActorMocked(msgs=[]))
+    monkeypatch.setattr(reporting, 'create_report', create_report_mocked())
+    monkeypatch.setattr(api, 'current_logger', logger_mocked())
+    reportleftoverpackages.process()
+
+    assert reporting.create_report.called == 0
+    assert api.current_logger.infomsg == ['No leftover packages, skipping...']
+
+
+def test_no_removed_packages(monkeypatch):
+    leftover_packages = LeftoverPackages(items=[RPM(name='rpm', version='1.0', release='1.el7', epoch='0',
+                                                    packager='foo', arch='noarch', pgpsig='SIG')])
+    monkeypatch.setattr(api, 'current_actor', CurrentActorMocked(msgs=[leftover_packages]))
+    monkeypatch.setattr(reporting, 'create_report', create_report_mocked())
+    monkeypatch.setattr(api, 'current_logger', logger_mocked())
+    reportleftoverpackages.process()
+
+    assert reporting.create_report.called == 1
+    assert 'Some RHEL packages have not been upgraded' in reporting.create_report.report_fields['title']
+    assert 'Following RHEL packages have not been upgraded' in reporting.create_report.report_fields['summary']
+    summary = 'Please remove these packages to keep your system in supported state.'
+    assert summary in reporting.create_report.report_fields['summary']
+
+
+def test_removed_packages(monkeypatch):
+    removed_packages = RemovedPackages(items=[RPM(name='rpm', version='1.0', release='1.el7', epoch='0',
+                                                  packager='foo', arch='noarch', pgpsig='SIG')])
+
+    monkeypatch.setattr(api, 'current_actor', CurrentActorMocked(msgs=[removed_packages]))
+    monkeypatch.setattr(reporting, 'create_report', create_report_mocked())
+    reportleftoverpackages.process()
+
+    assert reporting.create_report.called == 1
+    assert 'Leftover RHEL packages have been removed' in reporting.create_report.report_fields['title']
+    assert 'Following packages have been removed' in reporting.create_report.report_fields['summary']
+    assert 'rpm-1.0-1.el7' in reporting.create_report.report_fields['summary']
+
+
+def test_removed_packages_no_items(monkeypatch):
```

Comment:
Regarding the bug in the library, also update this test accordingly and add/update other test to test the correct behavior.

Reply:
Removed that one test, since when I refactored the code, the test was no longer needed and tested the same functionality as the other one test.